### ISSUE TITLE
[CI] FIPS FollowIndexIT testDowngradeRemoteClusterToBasic failure to recreate SSL context

### ISSUE THREAD

**Issue by albertzaharovits** [elastic/elasticsearch#52034]
This reproduces on master!
```
./gradlew ':x-pack:plugin:ccr:qa:downgrade-to-basic-license:follow-clusterRunner' --tests "org.elasticsearch.xpack.ccr.FollowIndexIT.testDowngradeRemoteClusterToBasic" \
  -Dtests.seed=FD9650072FA219B4 \
  -Dtests.security.manager=true \
  -Dtests.locale=nl \
  -Dtests.timezone=Europe/Zurich \
  -Dcompiler.java=13 \
  -Dtests.fips.enabled=true
```

```
        java.security.NoSuchAlgorithmException: Unable to invoke creator for DEFAULT: Default key/trust managers unavailable	
            at org.bouncycastle.jsse.provider.BouncyCastleJsseProvider$BcJsseService.newInstance(BouncyCastleJsseProvider.java:388)	
            at java.base/sun.security.jca.GetInstance.getInstance(GetInstance.java:236)	
            at java.base/sun.security.jca.GetInstance.getInstance(GetInstance.java:164)	
            at java.base/javax.net.ssl.SSLContext.getInstance(SSLContext.java:184)	
            at java.base/javax.net.ssl.SSLContext.getDefault(SSLContext.java:110)	
            at org.elasticsearch.client.RestClientBuilder.createHttpClient(RestClientBuilder.java:212)	
            ... 40 more	
            Caused by:	
            java.security.KeyManagementException: Default key/trust managers unavailable	
                at org.bouncycastle.jsse.provider.DefaultSSLContextSpi.<init>(DefaultSSLContextSpi.java:88)	
                at org.bouncycastle.jsse.provider.BouncyCastleJsseProvider$7.createInstance(BouncyCastleJsseProvider.java:217)	
                at org.bouncycastle.jsse.provider.BouncyCastleJsseProvider$BcJsseService.newInstance(BouncyCastleJsseProvider.java:373)	
                ... 45 more	
                Caused by:	
                java.security.AccessControlException: access denied ("java.io.FilePermission" "/dev/shm/elastic+elasticsearch+master+matrix-java-periodic-fips/ES_BUILD_JAVA/openjdk13/ES_RUNTIME_JAVA/openjdk14/nodes/general-purpose/x-pack/plugin/ccr/qa/downgrade-to-basic-license/build/build-tools-exported/cacerts.bcfks" "read")	
                    at java.base/java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)	
                    at java.base/java.security.AccessController.checkPermission(AccessController.java:1036)	
                    at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:408)	
                    at java.base/java.lang.SecurityManager.checkRead(SecurityManager.java:747)	
                    at java.base/java.io.File.exists(File.java:818)	
                    at org.bouncycastle.jsse.provider.ProvTrustManagerFactorySpi.getDefaultTrustStore(ProvTrustManagerFactorySpi.java:49)	
                    at org.bouncycastle.jsse.provider.ProvSSLContextSpi.getDefaultTrustManagers(ProvSSLContextSpi.java:302)	
                    at org.bouncycastle.jsse.provider.DefaultSSLContextSpi$LazyManagers.<clinit>(DefaultSSLContextSpi.java:57)	
                    at org.bouncycastle.jsse.provider.DefaultSSLContextSpi.<init>(DefaultSSLContextSpi.java:86)	
                    ... 47 more
```

Build scans:
https://gradle-enterprise.elastic.co/s/wfh66uffpg4hy
https://gradle-enterprise.elastic.co/s/ppmoxjtf2cvby
https://gradle-enterprise.elastic.co/s/b6kyaeu7kqrfy

**Comment by albertzaharovits** [comment]
Ooops, I just realized the name of the job `matrix-java-periodic-fips` and that the `FIPS` tests are not in a good mood today.

**Comment by albertzaharovits** [comment]
cc @jkakavas

**Comment by jkakavas** [comment]
Resolved in #52046
